modules/CodeCoverage.cmake

```diff
@@ -0,0 +1,683 @@
+# https://github.com/bilke/cmake-modules/blob/master/CodeCoverage.cmake
```

Comment:
Can we include this via submodule rather than copying it into our project?

Reply:
https://github.com/bilke/cmake-modules

It's an amalgamation of unrelated cmake files, I didn't want to pollute our cmake namespace with a bunch of irrelevant cmake files so I did a direct copy.

Reply:
Why would the unused ones polute our namespace? Can't we `include()` only the one we need?

Reply:
nvm I forgot about `include`, yeah it's fine to use a submodule, i put it in `extern`